modules/http/src/main/java/org/mule/module/http/internal/request/MuleEventToHttpRequest.java

```diff
@@ -74,7 +74,10 @@ public HttpRequestBuilder create(MuleEvent event, String resolvedMethod, String
 
         for (String outboundProperty : event.getMessage().getOutboundPropertyNames())
         {
-            builder.addHeader(outboundProperty, event.getMessage().getOutboundProperty(outboundProperty).toString());
+            if (!builder.getHeaders().containsKey(outboundProperty))
```

Comment:
What about this specific scenario? The test is the same in both branches and in 3.6.x it fails if I let outbound properties replace the headers defined in the request builder.